public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -262,14 +262,20 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
             standardConceptFilter = StandardConceptFilter.STANDARD_OR_CODE_ID_MATCH;
         }
 
+        List<String> domainIds = null;
+        if (searchConceptsRequest.getDomain() != null) {
+            domainIds = DOMAIN_MAP.get(searchConceptsRequest.getDomain()).asList();
+        }
+
         if(searchConceptsRequest.getQuery() == null || searchConceptsRequest.getQuery().isEmpty()){
             List<Concept> concepts;
+            standardConceptFilter = (standardConceptFilter==StandardConceptFilter.STANDARD_OR_CODE_ID_MATCH ? StandardConceptFilter.STANDARD_CONCEPTS : standardConceptFilter);
             if(standardConceptFilter == StandardConceptFilter.STANDARD_CONCEPTS){
-                concepts = conceptDao.findAllStandardConceptsOrderedByCount(maxResults);
+                concepts = conceptDao.findAllStandardConceptsOrderedByCount(maxResults, domainIds);
```

Comment:
Do we need to filter on anything besides domainIds here?

I'm wondering whether we could change the advanced search functionality to just make the filtering on keyword optional... then we might not need this whole if block.

Reply:
For now, the only filter is the domain ids.

Reply:
I don't follow. But we need to add the minCount to all of these functions right to implement that functionality ? 

This function is starting to get messy eh ? One thing we could do to simplify it is have a separate api call for findAll() .. And the uis could check for query string and call a different api to findAll.

Reply:
That sounds reasonable, it would make the searchConcepts api cleaner. I will do that.
Thanks.

Reply:
Added.